### PR TITLE
add opa user documentation

### PR DESCRIPTION
**Pull request by suchapalaver**
- References to OPA standard
- Explanation of OPA execution in chronicle - GraphQL & TP
- Cookbook style examples of authorization rules
- OPA transaction processor (`opa-tp`) and `opactl`

[CHRON-291](https://blockchaintp.atlassian.net/browse/CHRON-291)

Signed-off-by: Joseph Livesey [joseph.livesey@btp.works](mailto:joseph.livesey@btp.works)

[CHRON-291]: https://blockchaintp.atlassian.net/browse/CHRON-291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mtbc** [comment]
Great, thank you, I shall base the CHRON-292 text on top of your commits here.

**Comment by mtbc** [comment]
On reordering the CLI arguments, I'd been taking an opposite approach of clustering related ones. Let's start with your reordering here but I'll look at maybe breaking them into subsections? Perhaps that'll also happen to somehow fix the `text-transform: uppercase` styling issue.

File: docs/opa.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+Chronicle integrates OPA execution at its GraphQL API request point and in its Transaction
+Processor to ensure access control and policy compliance. OPA verifies user permissions
+against policies that define which actions are allowed and denied to whom. Integrating
+OPA execution in the transaction processor ensures that only transactions that meet
+specific criteria are accepted and processed by the network.
+
+## Authentication Rules in OPA
+
+Authentication rules in Open Policy Agent (OPA) determine whether a user or client
```
Comment (mtbc) [inline docs/opa.md:24]: I wonder if we may be conflating authentication and authorization here? I'd have thought the rules more regarded the latter but I may be unfamiliar with practitioners' current parlance.

File: docs/opa.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+is authorized to access a resource or perform an action.
+
+As an example of how to define an authentication rule in OPA, suppose we have a REST
+API endpoint that returns a list of users, but we want to restrict access to this
```
Comment (mtbc) [inline docs/opa.md:28]:
```suggestion
API endpoint that returns a list of users but we want to restrict access to this
```

File: docs/opa.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+endpoint only to users who have the "admin" role. We can define an OPA policy like
+this:
+
+```rego
+
```
Comment (mtbc) [inline docs/opa.md:33]: omit the blank line

File: docs/opa.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+The available options are:
+
+- `--connect` (`-C`): Sets the sawtooth validator address. This option takes a URL
+  as its argument.
+
+- `--completions`: Generate shell completions for the opa-tp command. This option
+  takes a shell type as its argument (either bash, zsh, or fish).
+
+- `--instrument`: Instrument the opa-tp process using the RUST_LOG environment variable.
```
Comment (mtbc) [inline docs/opa.md:105]:
```suggestion
- `--instrument`: Instrument the opa-tp process using the `RUST_LOG` environment variable.
```

File: docs/opa.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  This option takes a URL as its argument.
+
+- `--console-logging`: Enable console logging for `opa-tp` using the `RUST_LOG`
+  environment variable. This option takes a string argument, which can be set to
+  pretty or json to enable pretty-printed or JSON-formatted logging output, respectively.
```
Comment (mtbc) [inline docs/opa.md:110]:
```suggestion
  `pretty` or `json` to enable pretty-printed or JSON-formatted logging output, respectively.
```

Comment (mtbc) [inline docs/opa.md:110]:
```suggestion
  `pretty` or `json` to enable pretty-printed or JSON-formatted logging output, respectively.
```

File: docs/opa.md (hunk 1 above)
Comment (suchapalaver) [reply docs/opa.md:24]: Here, I was using the language used in the ticket.

Comment (mtbc) [reply docs/opa.md:24]: Ah, let's check with @ryanbtp then.

Comment (suchapalaver) [reply docs/opa.md]: When I think about it, I agree with you. So I've made the changes.

**Pull request merged**